src/ngraph/runtime/backend.hpp

```diff
@@ -133,8 +99,64 @@ class ngraph::runtime::Backend
     /// \brief Test if a backend particular property is supported
     /// \param prop is the feature to test.
     /// \returns true if the property is supported, false otherwise.
-    virtual bool is_supported_property(const Property prop) const { return false; }
-    void validate(std::shared_ptr<const Function> func,
-                  const std::vector<std::shared_ptr<runtime::Tensor>>& outputs,
+    virtual bool is_supported_property(const Property prop) const;
+
+    virtual void remove_compiled_function(std::shared_ptr<Executable> exec);
+
+    /// The following methods are temporary hacks to reduce the number of changes in this PR
+    /// They will be removed in a follow-on PR
+    bool call_with_validate(std::shared_ptr<Executable> handle,
+                            const std::vector<std::shared_ptr<runtime::Tensor>>& outputs,
+                            const std::vector<std::shared_ptr<runtime::Tensor>>& inputs);
+    bool call_with_validate(const std::unique_ptr<Executable>& handle,
+                            const std::vector<std::shared_ptr<runtime::Tensor>>& outputs,
+                            const std::vector<std::shared_ptr<runtime::Tensor>>& inputs);
+};
+
+class ngraph::runtime::Executable
```

Comment:
It would be great if you could add part of the PR description to this class as doxygen doc